pkg/prometheus/resource_selector.go

```diff
@@ -469,7 +474,7 @@ func (rs *ResourceSelector) SelectPodMonitors(ctx context.Context, listFn ListAl
 		}
 
 		if err = validateScrapeClass(rs.p, pm.Spec.ScrapeClassName); err != nil {
-			rejectFn(pm, err)
+			rejectFn(pm, fmt.Errorf("scrapeClass: %w", err))
```

Comment:
I don't think that wrapping the error is required here.

Reply:
In that case should I remove it from thé servicemonitor validation ?

Reply:
I removed it for all 4 places